src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -313,6 +313,10 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
                 {
                     isSuppressed = severity == ReportDiagnostic.Suppress;
                 }
+                else if (isEnabledWithAnalyzerConfigOptions(diag.Id, analyzerExecutor.Compilation))
```

Comment:
This file has the only product change, rest are all test changes.

Reply:
This looks wrong. Editorconfig should override compilation-wide settings.

Reply:
Also, can we share this logic with FilterDiagnostics somehow?

Reply:
> This looks wrong. Editorconfig should override compilation-wide settings.

This was done to match current compiler logic. If you have a suppression entry for any compiler warning OR enabled by default analyzer diagnostic in ruleset or via /nowarn, and an explicit entry in editorconfig that is not at none, the diagnostic is not reported.

Reply:
> Also, can we share this logic with FilterDiagnostics somehow?

I don't understand. This is an API invoked before making any analyzer callbacks to determine whether or not the analyzer should be completely suppressed. If suppressed, then analyzer never executes and cannot report diagnostics, so filtering is not a concern for these analyzers.

Reply:
> Editorconfig should override compilation-wide settings.

Note that having editorconfig override compilation-wide settings was my first implementation. But it caused inconsistency with the behavior cited here https://github.com/dotnet/roslyn/pull/38792#discussion_r328323377. I am fine with switching to those semantics and file a compiler bug to fix the inconsistency.

Reply:
Seems like the behavior cited in https://github.com/dotnet/roslyn/pull/38792#discussion_r328323377 only applies analyzer diagnostics, so it was likely a fallout from the same issue that IsDiagnosticAnalyzerSuppressed method was never updated to account for editorconfig overrides. I will fix that case as well.

Reply:
> Also, can we share this logic with FilterDiagnostics somehow?

By this I mean, logically what we are asking is: after applying all diagnostic settings, is the effective severity == Suppress, for the given location in source code. This seems isomorphic to the compiler filtering behavior. We should share the two.

Now that I know that the analyzer behavior had different logic from the compiler specifically *because* the two things are not shared, I'm now certain we should share the implementations.

Reply:
Compiler does filtering as a post-pass. IsDiagnosticAnalyzerSuppressed is executed prior to any analyzer callbacks (except fetching SupportedDiagnostics), especially designed as a performance hook to avoid running analyzers that are disabled in rulesets and nowarn.

Reply:
I understand we will use them for different purposes. That doesn't mean that the check is not isomorphic. Fundamentally, we want a central piece of logic to process compiler options and decide if a diagnostic is suppressed. That is the same, regardless of the point where we apply it.

Reply:
Filtering applies to specific diagnostic instances, which have location(s). These apply to both compiler + analyzer diagnostics.
The logic to avoid executing analyzers applies to the analyzers, not reported diagnostics.

I am not sure what part are you proposing to consolidate. Any case, feel free to file a separate issue for code refactoring as per your preference, this PR targets fixing a bug in existing implementation.